pallets/grants/src/lib.rs

```diff
@@ -216,16 +220,19 @@ pub mod pallet {
 			log::debug!("got new schedules for overwriting: {:#?}", new_schedules);
 
 			let target = T::Lookup::lookup(who)?;
-			VestingSchedules::<T>::insert(target.clone(), new_schedules.clone());
+
+			let new_schedules_bvec =
+				<BoundedVec<VestingScheduleOf<T>, T::MaxSchedule>>::try_from(new_schedules.clone())
+					.map_err(|_| <Error<T>>::MaxScheduleOverflow)?;
+
+			VestingSchedules::<T>::insert(target.clone(), new_schedules_bvec);
```

Comment:
If you set the `new_schedules` param to be a `BoundedVec` you should be able to avoid this.

Reply:
Indeed not relevant function `overwrite_vesting_schedules()` got removed.